Fix all warnings in frontend

Christdej opened the pull request:
Removing unused code, add callbacks and other minor changes.

GodVenn reviewed (commented):
Are you sure this works as expected? I have bad experiences with adding parameters in the useEffect parameter array

Christdej commented:
> Are you sure this works as expected? I have bad experiences with adding parameters in the useEffect parameter array

Yes, I have tested quite a bit locally atleast. And I agreed that there can be problems adding parameters in the useEffect parameter array, if you dont do it correctly. That is why I have added useCallback on most of the parameters in the parameters array, to try to ensure correct use, such that we dont create a new instance of an object every re-render of the object. I think there is only a few places I have added parameters to the array without adding callbacks or other measures to ensure it is not bugged
